pycpt/src/cptextras/view.py

```diff
@@ -5,43 +5,40 @@
 import xarray as xr
 import numpy as np
 
-def guess_coords_view_prob(X, x_lat_dim=None, x_lon_dim=None, x_sample_dim=None, x_feature_dim=None):
-	assert type(X) == xr.DataArray, 'X must be a data array'
-	common_x = ['LONGITUDE', 'LONG', 'X', 'LON']
-	common_y = ['LATITUDE', 'LAT', 'LATI', 'Y']
-	common_t = ['T', 'S', 'TIME', 'SAMPLES', 'SAMPLE', 'INITIALIZATION', 'INIT', "TARGET"]
-	common_m = ['M', 'FEATURES', 'F', 'REALIZATION', 'MEMBER', 'Z', 'C', 'CAT']
-	ret = {'lat': x_lat_dim, 'lon': x_lon_dim, 'samp': x_sample_dim, 'feat': x_feature_dim}
-	for dim in X.dims:
-		for x in common_x:
-			if x in dim.upper() and ret['lon'] is None:
-				ret['lon'] = dim
-		for y in common_y:
-			if y in dim.upper() and ret['lat'] is None:
-				ret['lat'] = dim
-		for t in common_t:
-			if t in dim.upper() and ret['samp'] is None:
-				ret['samp'] = dim
-		for m in common_m:
-			if m in dim.upper() and ret['feat'] is None:
-				ret['feat'] = dim
-	#assert None not in ret.values(), 'Could not detect one or more dimensions: \n  LATITUDE: {lat}\n  LONGITUDE: {lon}\n  SAMPLE: {samp}\n  FEATURE: {feat}\n'.format(**ret)
-	vals = []
-	for val in ret.values():
-		if val not in vals:
-			vals.append(val)
-	#assert len(vals) == 4, 'Detection Faild - Duplicated Coordinate: \n  LATITUDE: {lat}\n  LONGITUDE: {lon}\n  SAMPLE: {samp}\n  FEATURE: {feat}\n'.format(**ret)
-	return ret['lat'], ret['lon'], ret['samp'], ret['feat']
-
-
-def view_probabilistic(X, x_lat_dim=None, x_lon_dim=None, x_sample_dim=None, x_feature_dim=None, cmap_an=plt.get_cmap('cpt.pr_blue_r', 9), cmap_bn=plt.get_cmap('cpt.pr_red', 9), cmap_nn=plt.get_cmap("cpt.pr_green_r", 4), orientation='horizontal'):
-	x_lat_dim, x_lon_dim, x_sample_dim, x_feature_dim = guess_coords_view_prob(X, x_lat_dim, x_lon_dim, x_sample_dim, x_feature_dim)
-	assert x_sample_dim is None, 'View probabilistic requires you to select across sample dim to eliminate that dimension first'
-	#check_all(X, x_lat_dim, x_lon_dim, x_sample_dim, x_feature_dim)
-	assert x_lat_dim in X.coords.keys(), 'pycpt requires a dataset_lat_dim to be a coordinate on X'
-	assert x_lon_dim in X.coords.keys(), 'pycpt requires a dataset_lon_dim to be a coordinate on X'
-	assert x_feature_dim in X.coords.keys(), 'pycpt requires a dataset_feature_dim to be a coordinate on X'
-	assert type(X) == xr.DataArray, 'pycpt requires a dataset to be of type "Xarray.DataArray"'
+
+def view_coords_stations(domain,vals):
+    step=0.1
+    horizontal = len( np.arange(vals['X'].values.min(), vals['X'].values.max() + step, step) )
+    vertical = len( np.arange(vals['Y'].values.min(), vals['Y'].values.max() + step, step) )
+    
+    if horizontal>=vertical:
+       factor_percent_h=0.10
+       factor_percent_v=0.20
+    else:
+	   #50% seems like a lot in general, but it manages to cover Chile's borders well.
```

Comment:
@xchourio I think 50% would be excessive in the vast majority of cases, even countries that are vertically oriented. I think the shape of Chile is so unusual that we should just treat it as a special case, by using the explicit `domain` kwarg that you added.

I did that yesterday in https://github.com/iri-pycpt/notebooks/pull/15/commits/a993a682ff25d994721a39a58bb3c2db8ec866e8, but you didn't react to it, so maybe you didn't see it.

Reply:
BTW what's special about Chile is not only that it's tall and narrow, but also that there are no stations in the southwest coastal area, which leads to that part of the country getting cut off by the default cropping.

Reply:
> BTW what's special about Chile is not only that it's tall and narrow, but also that there are no stations in the southwest coastal area, which leads to that part of the country getting cut off by the default cropping.

I revoked my last changes.